modules/carto/src/api/maps-v3-client.ts

```diff
@@ -134,13 +136,27 @@ function getParameters({
   columns,
   clientId,
   aggregationExp,
-  aggregationResLevel
+  aggregationResLevel,
+  queryParameters
 }: Omit<FetchLayerDataParams, 'connection' | 'credentials'>) {
   const parameters = [encodeParameter('client', clientId || DEFAULT_CLIENT)];
 
   const sourceName = type === MAP_TYPES.QUERY ? 'q' : 'name';
   parameters.push(encodeParameter(sourceName, source));
 
+  if (queryParameters?.length) {
+    const queryParametersUrl = queryParameters
+      .map((param, i) => {
+        const params: string[] = [];
+        Object.keys(param).forEach(key => {
+          params.push(`queryParameters[${i}].${key}=${encodeURIComponent(param[key])}`);
```

Comment:
This will make the URL very long as we are needlessly include the `queryParameters` string many times. Can you see my comment here? https://github.com/visgl/deck.gl/issues/7149#issuecomment-1190309906